mbuild/compound.py

```diff
@@ -2425,12 +2425,23 @@ def spin(self, theta, around):
             The angle by which to rotate the Compound, in radians.
         around : np.ndarray, shape=(3,), dtype=float
             The axis about which to spin the Compound.
+        anchor : mb.Compound, optional, default=None (self)
```

Comment:
If you pass it a compound, it should spin around the center of mass of that compound. This way you don't have to specify particle/compound, since it'll still use the center of mass of a single bead compound (the synonym for a particle in mbuild). Also, what does the "(self)" mean? I'm inferring that this means the value of None really will use self. Not sure if this is standard to put in the function definition, but maybe could add a bit of uncertainty?

Reply:
so if None is provided for the anchor, the Compound will rotate around its own self.pos, which will be its center of mass, this option is for when you want rotate a compound but around one particular particle instead the center of mass (e.g., rotate a chain attached to a surface, but you want the particle attached to the surface to be fixed, or else that bond will be wonky)